### PR TITLE
#405 : Failing test showing that line endings are converted to CRLF

### PR DESCRIPTION
**Pull request by pombredanne**
This test will fail on Windows for now until #405  is fixed

### CODE REVIEW

**Comment by pombredanne** [comment]
and this fails alright as it should

**Comment by pombreda** [comment]
@audreyr  do not apply this for now. It needs to be refined first.
The line endings cannot be converted blindly to LF.... some windows files must have CRLF endings such as cmd/bat scripts.... therefore the approach should be instead to preserve the line endings of a file instead. A tad more tricky

**Comment by hackebrot** [comment]
Not too sure how to proceed on #405 and #406 tbh :worried:

**Comment by pydanny** [comment]
🔔  What's the status on this PR?

**Comment by pombredanne** [comment]
@pydanny no idea :) let me rebase this first!

**Comment by pombredanne** [comment]
This is rebased and should pass the tests.... Yet the issue is still there: CRLF should not be translated to LF. That said, I have lost a lot (or most) of my patience for these kind of Windows minutiae.

**Comment by lorengordon** [comment]
This is driving me nuts as well. I've converted all files in the cookiecutter template to LF, created a .gitattributes file forcing line endings to LF (both in the template repo and in the project repo), and still the project keeps getting created with CRLF line endings! Any bright ideas?

**Comment by theodesp** [comment]
@pombredanne There are some conflicts that need to be fixed for that if is to be merged 👍

**Comment by pombredanne** [comment]
@theodesp I rebassed... but the vcs test patterns have evolved quite bit since last year so it may not pass...  feel free to update as needed to make this mergeable.

**Comment by pombredanne** [comment]
@theodesp tests arer failing  ... make it pass if you want this in! It is not like I thoroughly enjoy keeping patches alive for 2.5 ;) so it is yours now or anyone else ;)

**Comment by pombredanne** [comment]
@theodesp basically the code context has changed too much... This would need to be redone from the start

**Comment by pombredanne** [comment]
@pydanny @audreyr IMHO this PR should be closed at it is and someone can restart something using it as a base if needed.

**Comment by insspb** [comment]
@ssbarnea Please do not close PRs silently without any resolution. 
This PR is related to #1159 and tests from here can be used there as second recheck.

File: tests/vcs/test_clone.py [hunk 1]
```diff
@@ -217,3 +217,27 @@ def test_clone_unknown_subprocess_error(mocker, clone_dir):
...
+    )
+    readme = open(os.path.join(repo_dir, 'README.rst'), 'rb').readlines()
+    first_line = readme[0]
+    assert first_line.endswith('\n')
+    assert not first_line.endswith('\r\n')
+    if os.path.isdir('cookiecutter-pypackage'):
+        utils.rmtree('cookiecutter-pypackage')
+
+
+def test_hg_clone_does_not_change_line_endings_on_win():
```
Comment (insspb) [inline tests/vcs/test_clone.py:234]: This is critical and unacceptable. External executable should not be used in tests (hg in this case)

File: tests/vcs/test_clone.py [hunk 2]
```diff
@@ -217,3 +217,27 @@ def test_clone_unknown_subprocess_error(mocker, clone_dir):
             clone_to_dir=clone_dir,
             no_input=True
         )
+
+
+def test_git_clone_does_not_change_line_endings_on_win():
+    repo_dir = vcs.clone(
```
Comment (insspb) [inline tests/vcs/test_clone.py:223]: This is critical and unacceptable. External executable should not be used in tests (git in this case)

**Comment by insspb** [comment]
Restored as #1407

File: tests/vcs/test_clone.py (hunk 1 above)
Comment (insspb) [reply tests/vcs/test_clone.py:234]: Fixed in #1407

File: tests/vcs/test_clone.py (hunk 2 above)
Comment (insspb) [reply tests/vcs/test_clone.py:223]: Fixed in #1407

Comment (pombredanne) [reply tests/vcs/test_clone.py:223]: @insspb you are correct ... and way back then, I might have been lazy to use mocks ;)
Thank you  for picking up the ball and I am glad this was of some use after all.

**Pull request closed without merging**